From: Eric Sandeen <sandeen@sandeen.net>
Date: Mon, 1 Feb 2010 10:13:36 -0600
Subject: [PATCH] xfsprogs: fix build with latest glibc headers

glibc in rawhide has some changes...

* Tue Jan 12 2010 Andreas Schwab <schwab@redhat.com> - 2.11.90-8
- Update from master.
  - More POSIX conformance fixes.

* Mon Jan 11 2010 Andreas Schwab <schwab@redhat.com> - 2.11.90-6
- Update from master.
  - POSIX conformance fixes (BZ#11125).

which seem to break the xfsprogs build.  I'm no feature test macro
guru, but the following gets it going again for me.

Signed-off-by: Eric Sandeen <sandeen@redhat.com>
Reviewed-by: Christoph Hellwig <hch@lst.de>
---
 include/builddefs.in |    2 +-
 include/linux.h      |    1 +
 2 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/include/builddefs.in b/include/builddefs.in
index ca8f172..cc75b5d 100644
--- a/include/builddefs.in
+++ b/include/builddefs.in
@@ -102,7 +102,7 @@ GCCFLAGS = -funsigned-char -fno-strict-aliasing -Wall
 #	   -Wbitwise -Wno-transparent-union -Wno-old-initializer -Wno-decl
 
 ifeq ($(PKG_PLATFORM),linux)
-PCFLAGS = -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 $(GCCFLAGS)
+PCFLAGS = -D_GNU_SOURCE -D_XOPEN_SOURCE=500 -D_FILE_OFFSET_BITS=64 $(GCCFLAGS)
 DEPENDFLAGS = -D__linux__
 endif
 ifeq ($(PKG_PLATFORM),darwin)
diff --git a/include/linux.h b/include/linux.h
index dbfb4cf..b342e55 100644
--- a/include/linux.h
+++ b/include/linux.h
@@ -22,6 +22,7 @@
 #include <sys/ioctl.h>
 #include <sys/param.h>
 #include <sys/sysmacros.h>
+#include <sys/stat.h>
 #include <malloc.h>
 #include <getopt.h>
 #include <endian.h>
-- 
1.7.2.3

